Automattic/wp-calypso
# Update Jetpack Pricing Page Banner

#### Changes proposed in this Pull Request

* Update "Discount & Money-back Guarantee" banner with a new, narrow design on Jetpack Pricing, WordPress.com, and Jetpack Connect Store pages
   * see design spec at iIYWCXVJSm06ZfzbeEWUT0-fi -> "I2 Designs [M3] - Update"
   * currently keep behind new Calypso flag, `jetpack/pricing-page-v2-banner`
* Move `useDetectWindowBoundry` hook to common Calypso lib

##### Screenshots

###### Wide page, banner non-stickied

| Jetpack Pricing | WordPress.com Plans | Jetpack Connect Store |
| --------------- | ------------------- | --------------------- |
| <img width="1181" alt="Screen Shot 2022-01-10 at 12 12 16" src="https://user-images.githubusercontent.com/2810519/148835312-f6625de8-5000-4751-9d1a-e15af20a0fdd.png"> | <img width="1181" alt="Screen Shot 2022-01-10 at 12 10 12" src="https://user-images.githubusercontent.com/2810519/148835296-e2fc3479-298f-47ef-a26a-32abb46f06a5.png"> |<img width="1181" alt="Screen Shot 2022-01-10 at 12 10 54" src="https://user-images.githubusercontent.com/2810519/148835376-9b63d5f9-dff9-440e-bea3-67aa6bc18071.png"> |


###### Wide page, banner stickied

| Jetpack Pricing | WordPress.com Plans | Jetpack Connect Store |
| --------------- | ------------------- | --------------------- |
|<img width="1181" alt="Screen Shot 2022-01-10 at 12 12 21" src="https://user-images.githubusercontent.com/2810519/148835438-6585eeea-d940-4792-a46a-3bad6502efd4.png"> | <img width="1181" alt="Screen Shot 2022-01-10 at 12 11 10" src="https://user-images.githubusercontent.com/2810519/148835543-4ebecb62-89ac-429c-acba-5c98339cb937.png"> |<img width="1181" alt="Screen Shot 2022-01-10 at 12 11 21" src="https://user-images.githubusercontent.com/2810519/148835584-f949eac5-e510-4c01-bdb0-e04b83191900.png">

###### Narrow page, banner non-stickied

| Jetpack Pricing | WordPress.com Plans | Jetpack Connect Store |
| --------------- | ------------------- | --------------------- |
| <img width="453" alt="Screen Shot 2022-01-10 at 12 12 38" src="https://user-images.githubusercontent.com/2810519/148835680-3b6b1609-fd3f-4f4e-8269-537075fda01a.png"> | <img width="453" alt="Screen Shot 2022-01-10 at 12 13 17" src="https://user-images.githubusercontent.com/2810519/148835726-f829c5ca-6233-4391-b7f6-63b56adcec68.png"> |<img width="453" alt="Screen Shot 2022-01-10 at 12 13 08" src="https://user-images.githubusercontent.com/2810519/148835765-9466935e-7edc-44e3-93b7-ba7254f85cc6.png"> |

###### Narrow page, banner stickied

| Jetpack Pricing | WordPress.com Plans | Jetpack Connect Store |
| --------------- | ------------------- | --------------------- |
| <img width="453" alt="Screen Shot 2022-01-10 at 12 13 03" src="https://user-images.githubusercontent.com/2810519/148835896-f33fa346-f25e-4841-8048-95fd47740a8b.png"> |<img width="453" alt="Screen Shot 2022-01-10 at 12 12 57" src="https://user-images.githubusercontent.com/2810519/148835925-91b3f55e-dd20-4ad3-9d94-42b2e486b0b1.png"> |<img width="453" alt="Screen Shot 2022-01-10 at 12 13 03" src="https://user-images.githubusercontent.com/2810519/148835959-db7142c9-d0a3-458d-b6f6-12b0eddc7fe7.png"> |


#### Testing instructions

1. Navigate to each page on the branch:
 - **(Calypso Green)** Jetpack Pricing `jetpack.cloud.localhost:3000/pricing`
 - **(Calypso Blue)** WordPress.com Plans `calypso.localhost:3000/plans/:siteSlug`
 - **(Calypso Blue)** Jetpack Connect Store `calypso.localhost:3000/jetpack/connect/store`
2. Verify that the page is the same as `trunk` with minor changes to spacing
3. Add the flag to the URL: `?flags=jetpack/pricing-page-v2-banner` ( if #59914 is unmerged, use `?flags=jetpack/pricing-page-v2-banner,jetpack/pricing-page-annual-only` instead )
4. Verify the banner switches to the new compact design
5. Verify the banner is sticky and compacts to only show the "50% off" copy at narrow widths.


Related to 1201295898168937-as-1201295899380976
